### PR TITLE
Bugfix. Spaces are not inserted after { and before } in lambda expressions (#711)

### LINKED ISSUE #711: Spaces are not inserted after { and before } in lambda expressions
## Describe the bug
```kotlin
fun foo() {
    list.map {it.text}
}
```
should be fixed to

```kotlin
fun foo() {
    list.map { it.text }
}
```
but it isn't.

## Environment information
* diktat version: 0.3.0

### PR DESCRIPTION
**Pull request by aktsay6**
### What's done:
  * Fixed bugs
  * Added more tests

This pull request closes #711

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt [hunk 1]
```diff
@@ -160,10 +164,18 @@ class WhiteSpaceRule(private val configRules: List<RulesConfig>) : Rule("horizon
         }
     }
 
+    private fun handleRbrace(node: ASTNode) {
+        if (node.treeParent.elementType == FUNCTION_LITERAL && !node.treePrev.isWhiteSpace()) {
+            WRONG_WHITESPACE.warnAndFix(configRules, emitWarn, isFixMode, "there should be a whitespace before }", node.startOffset, node) {
+                node.treeParent.addChild(PsiWhiteSpaceImpl(" "), node)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt:170]: Are you sure this white space normally belongs to `FUNTION_LITERAL` and not to `BLOCK`? I remember we had some issues with something similar in the past.

Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt:170]: I checked the tree of the correct code and it was like this.

Comment (aktsay6) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt:170]: I mean the `WHITE_SPACE` was right after `FUNCTION_LITERAL`

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/WhiteSpaceRule.kt:170]: kk, good

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt [hunk 2]
```diff
@@ -50,7 +50,7 @@ class KotlinParser {
             }
         }  // I don't really understand what's going on here, but thanks to this, you can use this node in the future
         val project = KotlinCoreEnvironment.createForProduction(
-            Disposable {},
+            Disposable { },
```
Comment (aktsay6) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/KotlinParser.kt:53]: @petertrr I guess that's the case you were talking about. If the body is empty, then the `WHITE_SPACE` comes before `BLOCK` and not before `RBRACE`. Added test on this case.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/spaces/WhiteSpaceRuleWarnTest.kt [hunk 3]
```diff
@@ -615,4 +616,44 @@ class WhiteSpaceRuleWarnTest : LintTestBase(::WhiteSpaceRule) {
...
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.WRONG_WHITESPACE)
+    fun `should not trigger on braces with empty body`() {
+        lintMethod(
+            """
+                |val project = KotlinCoreEnvironment.createForProduction(
+                |   Disposable { },
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/spaces/WhiteSpaceRuleWarnTest.kt:652]: Please add also a test when there is no space: `{}`.

**Review by petertrr — APPROVED** [review]

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/spaces/WhiteSpaceRuleWarnTest.kt (hunk 3 above)
Comment (aktsay6) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/spaces/WhiteSpaceRuleWarnTest.kt:666]: Added

**Pull request merged**